infra/indexer/components/indexer.ts

```diff
@@ -0,0 +1,376 @@
+    });
+
+    const authToken = aws.ecr.getAuthorizationTokenOutput({
+      registryId: ecrRepository.repository.registryId,
+    });
+
+    // Optionally add in the gh token secret and sccache s3 creds to the build ctx
+    let buildSecrets = {};
+    if (ciCacheSecret !== undefined) {
+      const cacheFileData = ciCacheSecret.apply((filePath: any) => fs.readFileSync(filePath, 'utf8'));
+      buildSecrets = {
+        ci_cache_creds: cacheFileData,
+      };
+    }
+    if (githubTokenSecret !== undefined) {
+      buildSecrets = {
+        ...buildSecrets,
+        githubTokenSecret
+      }
+    }
+
+    const image = new docker_build.Image(`${serviceName}-img`, {
+      tags: [pulumi.interpolate`${ecrRepository.repository.repositoryUrl}:${dockerTag}`],
+      context: {
+        location: dockerDir,
+      },
+      platforms: ['linux/amd64'],
+      push: true,
+      dockerfile: {
+        location: `${dockerDir}/dockerfiles/indexer.dockerfile`,
+      },
+      buildArgs: {
+        S3_CACHE_PREFIX: 'private/boundless/rust-cache-docker-Linux-X64/sccache',
+      },
+      secrets: buildSecrets,
+      cacheFrom: [
+        {
+          registry: {
+            ref: pulumi.interpolate`${ecrRepository.repository.repositoryUrl}:cache`,
+          },
+        },
+      ],
+      cacheTo: [
+        {
+          registry: {
+            mode: docker_build.CacheMode.Max,
+            imageManifest: true,
+            ociMediaTypes: true,
+            ref: pulumi.interpolate`${ecrRepository.repository.repositoryUrl}:cache`,
+          },
+        },
+      ],
+      registries: [
+        {
+          address: ecrRepository.repository.repositoryUrl,
+          password: authToken.apply((authToken) => authToken.password),
+          username: authToken.apply((authToken) => authToken.userName),
+        },
+      ],
+    });
+
+    const indexerSecGroup = new aws.ec2.SecurityGroup(`${serviceName}-sg`, {
+      name: `${serviceName}-sg`,
+      vpcId: vpcId,
+      egress: [
+        {
+          fromPort: 0,
+          toPort: 0,
+          protocol: '-1',
+          cidrBlocks: ['0.0.0.0/0'],
+          ipv6CidrBlocks: ['::/0'],
+        },
+      ],
+    });
+
+    const rdsUser = 'indexer';
+    const rdsPort = 5432;
+    const rdsDbName = 'indexer';
+
+    const dbSubnets = new aws.rds.SubnetGroup(`${serviceName}-dbsubnets`, {
+      subnetIds: privSubNetIds,
+    });
+
+    const rdsSecurityGroup = new aws.ec2.SecurityGroup(`${serviceName}-rds`, {
+      name: `${serviceName}-rds`,
+      vpcId: vpcId,
+      ingress: [
+        {
+          fromPort: rdsPort,
+          toPort: rdsPort,
+          protocol: 'tcp',
+          securityGroups: [indexerSecGroup.id],
+        },
+      ],
+      egress: [
+        {
+          fromPort: 0,
+          toPort: 0,
+          protocol: '-1',
+          cidrBlocks: ['0.0.0.0/0'],
+        },
+      ],
+    });
+
+    const rds = new aws.rds.Instance(`${serviceName}-rds`, {
```

Comment:
Did you look into Aurora? Would be good if we don't have to worry about ops/scaling this up